### PR TITLE
Implement CaptureListener methods to collect capture data

### PR DESCRIPTION
**Pull request by kisenshi**
Implemented CaptureListener methods in ClientGgp so the data related to the capture is collected.

The data that is necessary to collect for the capture is: CaptureData, a vector containing the timers (TimerInfo) and a string manager.

OnCaptureComplete is also updated to set SamplingProfiler data when the capture has finished.

### CODE REVIEW

File: OrbitClientGgp/ClientGgp.cpp [hunk 1]
```diff
@@ -234,16 +242,39 @@ void ClientGgp::OnCaptureStarted(
...
+  SamplingProfiler sampling_profiler(*capture_data_.GetCallstackData(), capture_data_);
+  capture_data_.set_sampling_profiler(sampling_profiler);
+}
 
-void ClientGgp::OnTimer(const orbit_client_protos::TimerInfo&) {}
+void ClientGgp::OnTimer(const orbit_client_protos::TimerInfo& timer_info) {
+  if (timer_info.function_address() > 0) {
+    FunctionInfo* func =
+        capture_data_.process()->GetFunctionFromAddress(timer_info.function_address());
+    CHECK(func != nullptr);
```
Comment (danielfenner) [inline OrbitClientGgp/ClientGgp.cpp:255]: Is the Check-fail intended here? I briefly looked into GetFunctionFromAddress and it seems returning nullptr is not that far fetched. 
If all these cases are "impossible" (Is this the case? We only can get samples with symbols from our instrumentation?) add a brief comment in front of the CHECK.

Comment (florian-kuebler) [reply OrbitClientGgp/ClientGgp.cpp:255]: For function timers, the function *must* be present in the process. If this is not the case, our data model has changed, thus failing is good here.

**Review by IrinaShkviro — APPROVED** [review]
lgtm, thanks!

File: OrbitClientGgp/ClientGgp.cpp (hunk 1 above)
Comment (kisenshi) [reply OrbitClientGgp/ClientGgp.cpp:255]: I'm gonna add a brief comment before the check to clarify it, thanks!

Comment (kisenshi) [reply OrbitClientGgp/ClientGgp.cpp:256]: Done!

Comment (florian-kuebler) [reply OrbitClientGgp/ClientGgp.cpp:256]: Thanks!

**Review by danielfenner — APPROVED** [review]
LGTM!

**Comment by kisenshi** [comment]
Merging this, thanks!

**Pull request merged**